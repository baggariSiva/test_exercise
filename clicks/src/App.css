.App{
    text-align: center;
}


.Number{
  margin:5%;
  font-family: Roboto-Bold;
    font-size: 84px;
    color: #37474F;
    
  }
  .Button button{
    
   
  background: #5E19F5;
  border: 1px solid #979797;
  box-sizing: border-box;
  border-radius: 8px;
  padding:10px;
  width:250px;
  font-family: Roboto-Regular;
  font-size: 32px;
  color: #FFFFFF;
  }
  