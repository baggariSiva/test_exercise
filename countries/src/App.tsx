import React from "react";
import Countries from "components/CountriesData";
import {ApolloClient, ApolloProvider, InMemoryCache} from "@apollo/client";
const client = new ApolloClient({
    cache: new InMemoryCache(),
    uri: "https://countries.trevorblades.com"
});
const App: React.FC = () => {
    return (
        <ApolloProvider client={client}>
            <Countries />
        </ApolloProvider>
    );
};
export default App;
