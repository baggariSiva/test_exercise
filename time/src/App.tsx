import React from "react";
import Time from "./components/Time/Timer";

const App: React.FC = () => <Time />;

export default App;
