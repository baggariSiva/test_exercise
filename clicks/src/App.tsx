import React from "react";
import "./components/App.css";
import {inject, IWrappedComponent, observer} from "mobx-react";
import {CounterStore} from "./counterStore";

type StoreProps = {
    counterStore: CounterStore;
};

interface Props extends StoreProps {
    title: string;
}

@inject("counterStore")
@observer
class App extends React.Component<Props> {
    static defaultProps = {} as StoreProps;
    constructor(props) {
        super(props);
    }

    render() {
        const {counterStore, title} = this.props;

        return (
            <div className="App">
                <div>{title}</div>
                <div className="Number">Clicks: {counterStore.counter}</div>
                {console.log("counter", counterStore)}
                {console.log("props", this.props)}
                <div className="Button">
                    <span>
                        <button style={{marginRight: 100}} onClick={() => counterStore.increment()}>
                            Increment
                        </button>
                    </span>
                    <span>
                        {" "}
                        <button onClick={() => counterStore.decrement()}>Decrement</button>
                    </span>
                </div>
            </div>
        );
    }
}

export default App as typeof App & IWrappedComponent<Props>;
