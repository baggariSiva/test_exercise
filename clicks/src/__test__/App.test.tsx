import * as React from "react";
import App from "../App";
import {Provider} from "mobx-react";
import CounterStore from "../counterStore";
import renderer from "react-test-renderer";

it('App renders correctly', () => {
  const tree = renderer
    .create(<Provider counterStore={CounterStore}>
      <App title="" />
  </Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

   