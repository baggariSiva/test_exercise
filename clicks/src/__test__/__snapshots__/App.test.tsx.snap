// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="sc-bdnxRM jYRkKf"
  >
    <div>
      
    </div>
     
    <h2>
      Clicks: 
    </h2>
    <p>
      <span>
        <button
          onClick={[Function]}
          style={
            Object {
              "marginRight": 100,
            }
          }
        >
          Increment
        </button>
      </span>
      <span>
         
        <button
          onClick={[Function]}
        >
          Decrement
        </button>
      </span>
    </p>
  </div>
</div>
`;
